com.unity.ml-agents/Runtime/Inference/BarracudaModelExtensions.cs

```diff
@@ -249,7 +262,19 @@ public static int DiscreteOutputSize(this Model model)
             else
             {
                 var discreteOutputShape = model.GetTensorByName(TensorNames.DiscreteActionOutputShape);
-                return discreteOutputShape == null ? 0 : (int)discreteOutputShape[0];
+                if (discreteOutputShape == null)
+                {
+                    return 0;
+                }
+                else
+                {
+                    int result = 0;
```

Comment:
This is a behavior change for legacy models, right? Is it correct?

Reply:
It is not. (Unless I messed it up)
Because discreteOutputShape is currently a tensor of shape 1 ([sum_of_branches]) and will be changed to be ([branh0_size, branch1_size, ...]). So doing the sum does the same thing wether it is legacy or new.

Reply:
OK, thanks. Could you leave that as a comment here for the future readers?